### DISCUSSION TITLE
Gemini support

### DISCUSSION THREAD

**Discussion by dio-rz** [frankroeder/parrot.nvim#25] [Ideas]
Hi,

I want to ask you if you are open to adding Gemini support into the plugin? Looking this [PR](https://github.com/frankroeder/parrot.nvim/pull/20) I think I can create a draft PR with most of those changes. 

The only one I really don't know how to implement is the equivalent to [this file](https://github.com/frankroeder/parrot.nvim/pull/20/files#diff-669d2753f0edd28995e287e54ca14f1ffd93baa2394da640a15c2e33981e0b52), so I would really appreciate if you could help me with hints on how to properly implement that part.

Thanks a lot for your fork, I'm amazed for how good it works!

**Comment by frankroeder** [comment]
Hi @Dioprz, 

that is definitely possible. I am more than happy to help, and it shouldn't be too hard. Indeed, it is basically copying the mistral provider and applying the appropriate changes. dkuku provided me actually with an API key so that I can debug it before the final merge. Just let me know when you need help with it. Currently, I don't have a Gemini access.

I’m pleased to hear from you and appreciate your feature request!

**Comment by dio-rz** [comment]
Hi @frankroeder. Thanks a lot for such a warm welcome!

I will send you a draft in some minutes with the changes I was able to do. Let's continue the conversation there!
